joomla/joomla-cms
# [4.0][NO CACHE] Refactor the build tools pt.2

Pull Request for Issue # .

### Summary of Changes
The second part of the refactoring of the npm build tools.
- Templates get an ~~`src`~~ `assets_source` folder which can have subfolders for `css`, `js` or `scss`
- Scss files compiled (unmagnified and minified as well) and stored in the usual `template/css`
- CSS files get copied over to the usual `template/css` folder and minified as well 
- JS files can be any of the 3 versions already introduced in the previous PR (.es6.js, .es5.js and .w-c.es6.js) The respective product of these will be placed in `template/js`

- Watch gets the second round of revamp also it gets more specific.
- Without any param will watch the `build/media_source` folder
- Can also accept `administrator/templates` so someone can work on a backend template
- or `templates` and this will watch the front end templates
- ATM it will not handle plain css files (that will be on the next PR)
- Also there is no logic to synchronise the removal of a source file to be correctly reflected in the distribution folders (again that's for the next PR)
 
### Testing Instructions
- apply patch 
- run `npm init`
- run `node run build.js --watch`

edit the file `joomla-cms/build/media_source/plg_installer_webinstaller/scss/client.scss` check if the css and .min.css in the media forder was updated

Create a file `joomla-cms/build/media_source/plg_installer_webinstaller/scss/test.scss` with any valid code (eg body {display:none;} ) check if the css and .min.css in the media forder was created

press ctrl+c to quit the watch.

Repeat the procedure but this time use `node run build.js --watch administrator/templates` and edit create a file in the atum/src/scss folder

Happy coding 🍸

### Expected result



### Actual result



### Documentation Changes Required

Based on the https://github.com/joomla/joomla-cms/pull/23723
